src/preprocessor/preprocessor.ts

```diff
@@ -203,10 +203,16 @@ const expandFunctionMacro = (
     const argIdentifiers = macroArgs.map(
       (a) => (a as PreprocessorIdentifierNode).identifier
     );
+
     const argKeys = argIdentifiers.reduce<Record<string, string>>(
       (acc, identifier, index) => ({
         ...acc,
-        [identifier]: args[index].trim(),
+        // We are scanning from the outside in - so fn(args) - we need to
+        // recursivey test if args itself has macros to expand. We don't need
+        // to remove the current macroName from macros because we are scanning
+        // outside in, so self-referencing macros won't get into an infinite
+        // loop here
+        [identifier]: expandMacros(args[index].trim(), macros),
```

Comment:
This was not from the user reported bug directly - but a scan was missing here for arg replacement. The call to expandMacros() below does not catch this case because that one ignores self-referencing macros